src/dune_rules/coq/coq_lib.ml

```diff
@@ -407,6 +413,150 @@ module DB = struct
       Resolve.Memo.List.map theories ~f:(resolve ~coq_lang_version db)
     in
     Resolve.O.(theories >>= top_closure)
+
+  let empty_db =
+    let resolve _ = `Not_found in
+    { parent = None; resolve; boot = None }
+
+  let stdlib_lib ~coqlib =
+    let theories_dir =
+      Path.append_local coqlib (Path.Local.of_string "theories")
+    in
+    Memo.return
+    @@ { loc = Loc.none
+       ; boot = Resolve.return None
```

Comment:
actually we need to set this to some for the stdlib, don't we?

Reply:
The "boot" here only works when we have a workspace boot library. We need to clear up all our uses of boot anyway since we have at least 3 subtley different meanings present in the code.

Reply:
Boot needs to point to the stdlib installed globally as we need to do dep injection for prelude.vo.

So actually we are now always in "Boot" mode, except that we may have a stdlib or not, the latter for things like stdlib2, hott, etc...

Reply:
We also need to allow the case of having multiple boot libraries in a workspace. Say a few versions of Coq.

Reply:
How would that work tho? As of now, if we have the same (public) theory in a workspace we fail (same for OCaml right?)

Reply:
Yes, there are lots of caveats so we need discussion of the intended semantics at a later date.